frontend/src/metabase/ui/components/feedback/Loader/Loader.unit.spec.tsx

```diff
@@ -0,0 +1,54 @@
+import { MantineProvider } from "@mantine/core";
+import { render, screen } from "@testing-library/react";
+
+import { Loader, setCustomLoader } from "./Loader";
+
+const renderWithProvider = (component: React.ReactElement) => {
+  return render(<MantineProvider>{component}</MantineProvider>);
+};
+
+describe("Loader", () => {
+  beforeEach(() => {
+    setCustomLoader(undefined);
+  });
+
+  it("renders default Mantine Loader when no custom loader is set", () => {
+    renderWithProvider(<Loader data-testid="loader" />);
+
+    const loader = screen.getByTestId("loader");
+    expect(loader).toBeInTheDocument();
+    expect(loader).toHaveClass("mantine-Loader-root");
+  });
+
+  it("renders custom loader when one is set", () => {
+    const CustomLoader = () => (
+      <div data-testid="custom-loader">Custom Loader</div>
+    );
+
+    setCustomLoader(CustomLoader);
+    renderWithProvider(<Loader />);
+
+    expect(screen.getByTestId("custom-loader")).toBeInTheDocument();
```

Comment:
nit: I'd just use `getByText`, like we're literally set the loader text here a few lines above this assertion.